lib/MusicBrainz/Server/Controller/Collection.pm

```diff
@@ -92,9 +92,10 @@ sub show : Chained('load') PathPart('') {
     my $order = $c->req->params->{order};
 
     my $model = $c->model(type_to_model($entity_type));
-    my $entities = $self->_load_paged($c, sub {
-        $model->find_by_collection($collection->id, shift, shift, $order);
+    my $collection_items = $self->_load_paged($c, sub {
+        $model->find_items_by_collection($collection->id, shift, shift, $order);
     });
+    my $entities = map { $_->entity } @$collection_items;
```

Comment:
That'll assign the size of the array, so you instead need:
```suggestion
    my $entities = [map { $_->entity } @$collection_items];
```
I would've suggested just `my @entities = ...` but it uses `@$entities` like a hundred times below.